resources/assets/js/annotations/components/annotationCanvas.vue

```diff
@@ -515,44 +515,41 @@ export default {
             return this.convertPointsFromOlToDb(points);
         },
         handleNewFeature(e) {
-            if (this.hasSelectedLabel) {
-                let geometry = e.feature.getGeometry();
-                if (geometry.getType() === 'Polygon') {
-                    if (PolygonValidator.isInvalidPolygon(e.feature)) {
-                        this.$emit('is-invalid-polygon');
-                        // This must be done in the change event handler.
-                        // Not exactly sure why.
-                        this.annotationSource.once('change', () => {
-                            if (this.annotationSource.hasFeature(e.feature)) {
-                                this.annotationSource.removeFeature(e.feature);
-                            }
-                        });
-                        return;
+            if (!this.hasSelectedLabel || this.isInvalidShape(e.feature)) {
+                // This must be done in the change event handler.
+                // Not exactly sure why.
+                this.annotationSource.once('change', () => {
+                    if (this.annotationSource.hasFeature(e.feature)) {
+                        this.annotationSource.removeFeature(e.feature);
                     }
+                });
+                this.$emit('is-invalid-shape', e.feature.getGeometry().getType());
+                return;
+            }
```

Comment:
There should be no "invalid shape" error if no label is selected. This can be handled separately:

```suggestion
            if (!this.hasSelectedLabel) {
                this.annotationSource.removeFeature(e.feature);
                return;
            }
            
            if (this.isInvalidShape(e.feature)) {
                // This must be done in the change event handler.
                // Not exactly sure why.
                this.annotationSource.once('change', () => {
                    if (this.annotationSource.hasFeature(e.feature)) {
                        this.annotationSource.removeFeature(e.feature);
                    }
                });
                this.$emit('is-invalid-shape', e.feature.getGeometry().getType());
                return;
            }
```